evennia/contrib/turnbattle/tb_items.py

```diff
@@ -0,0 +1,1374 @@
+NONCOMBAT_TURN_TIME = 30 # Time per turn count out of combat
+
+"""
+----------------------------------------------------------------------------
+COMBAT FUNCTIONS START HERE
+----------------------------------------------------------------------------
+"""
+
+def roll_init(character):
+    """
+    Rolls a number between 1-1000 to determine initiative.
+
+    Args:
+        character (obj): The character to determine initiative for
+
+    Returns:
+        initiative (int): The character's place in initiative - higher
+        numbers go first.
+
+    Notes:
+        By default, does not reference the character and simply returns
+        a random integer from 1 to 1000.
+
+        Since the character is passed to this function, you can easily reference
+        a character's stats to determine an initiative roll - for example, if your
+        character has a 'dexterity' attribute, you can use it to give that character
+        an advantage in turn order, like so:
+
+        return (randint(1,20)) + character.db.dexterity
+
+        This way, characters with a higher dexterity will go first more often.
+    """
+    return randint(1, 1000)
+
+
+def get_attack(attacker, defender):
+    """
+    Returns a value for an attack roll.
+
+    Args:
+        attacker (obj): Character doing the attacking
+        defender (obj): Character being attacked
+
+    Returns:
+        attack_value (int): Attack roll value, compared against a defense value
+            to determine whether an attack hits or misses.
+
+    Notes:
+        This is where conditions affecting attack rolls are applied, as well.
+        Accuracy Up and Accuracy Down are also accounted for in itemfunc_attack(),
+        so that attack items' accuracy is affected as well.
+    """
+    # For this example, just return a random integer up to 100.
+    attack_value = randint(1, 100)
+    # Add 25 to the roll if the attacker has the "Accuracy Up" condition.
+    if "Accuracy Up" in attacker.db.conditions:
+        attack_value += 25
+    # Subtract 25 from the roll if the attack has the "Accuracy Down" condition.
+    if "Accuracy Down" in attacker.db.conditions:
+        attack_value -= 25
+    return attack_value
+
+
+def get_defense(attacker, defender):
+    """
+    Returns a value for defense, which an attack roll must equal or exceed in order
+    for an attack to hit.
+
+    Args:
+        attacker (obj): Character doing the attacking
+        defender (obj): Character being attacked
+
+    Returns:
+        defense_value (int): Defense value, compared against an attack roll
+            to determine whether an attack hits or misses.
+
+    Notes:
+        This is where conditions affecting defense are accounted for.
+    """
+    # For this example, just return 50, for about a 50/50 chance of hit.
+    defense_value = 50
+    # Add 15 to defense if the defender has the "Defense Up" condition.
+    if "Defense Up" in defender.db.conditions:
```

Comment:
Out of curiosity, why does the `Defense Up` Attribute not itself store the value `15`? That way a given defender could have different defense value from another. It sounds quite logical to store such things with the object.

Reply:
I thought about this, but decided that having different instances of the same conditions with varying levels of severity is an exercise best left to the developer, since different rules systems handle it so differently.